tiledb/CMakeLists.txt

```diff
@@ -437,30 +434,47 @@ if (TILEDB_STATIC)
   include(TileDBCommon)
 
   # Helper to generate CMake code
-  macro(append_dep_lib LIB_TARGET)
-    if (TARGET ${LIB_TARGET})
-      get_installed_location(LOC ${LIB_TARGET})
+  macro(append_dep_lib APPEND_LIB)
+    if (TARGET ${APPEND_LIB})
+      get_installed_location(TARGET_LOC ${APPEND_LIB})
+      get_target_property(TARGET_LIBS ${APPEND_LIB} INTERFACE_LINK_LIBRARIES)
+
+      string(CONCAT TILEDB_STATIC_DEP_STRING
+        "${TILEDB_STATIC_DEP_STRING}"
+        "if (NOT TARGET ${APPEND_LIB})\n"
+        "  add_library(${APPEND_LIB} UNKNOWN IMPORTED)\n"
+        "  set_target_properties(${APPEND_LIB} PROPERTIES IMPORTED_LOCATION ${TARGET_LOC})\n"
+      )
+
+      if (TARGET_LIBS AND ("${APPEND_LIB}" MATCHES "CURL"))
```

Comment:
@tdenniston @Shelnutt2 forwarding the link targets here was required to get the behavior required in https://github.com/TileDB-Inc/TileDB/issues/1080, however I made it curl-specific for now because it broke the build w/ capnproto.